src/fuelTanks/CCPACSVessel.cpp

```diff
@@ -0,0 +1,700 @@
+        throw tigl::CTiglError("Guide Curve with UID " + uid + " does not exists", TIGL_ERROR);
+    }
+    else {
+        throw CTiglError(_vesselTypeException);
+    }
+}
+
+std::vector<gp_Pnt> CCPACSVessel::GetGuideCurvePoints() const
+{
+    if (m_segments_choice1) {
+        std::vector<gp_Pnt> points;
+
+        // connect the belonging guide curve segments
+        for (int isegment = 1; isegment <= GetSegmentCount(); ++isegment) {
+            const CCPACSFuselageSegment& segment = m_segments_choice1.get().GetSegment(isegment);
+
+            if (!segment.GetGuideCurves()) {
+                continue;
+            }
+
+            const CCPACSGuideCurves& segmentCurves = *segment.GetGuideCurves();
+            for (int iguide = 1; iguide <= segmentCurves.GetGuideCurveCount(); ++iguide) {
+                const CCPACSGuideCurve& curve = segmentCurves.GetGuideCurve(iguide);
+                std::vector<gp_Pnt> curPoints = curve.GetCurvePoints();
+                points.insert(points.end(), curPoints.begin(), curPoints.end());
+            }
+        }
+        return points;
+    }
+    else {
+        throw CTiglError(_vesselTypeException);
+    }
+}
+
+// get short name for loft
+std::string CCPACSVessel::GetShortShapeName() const
+{
+    unsigned int findex = 0;
+    unsigned int i      = 0;
+
+    for (auto& v : GetParent()->GetVessels()) {
+        ++i;
+        if (GetUID() == v->GetUID()) {
+            findex = i;
+            std::stringstream shortName;
+            shortName << "V" << findex;
+            return shortName.str();
+        }
+    }
+    return "UNKNOWN";
+}
+
+void CCPACSVessel::SetFaceTraits(PNamedShape loft) const
+{
+    int nFacesTotal = GetNumberOfFaces(loft->Shape());
+    int nFacesAero  = nFacesTotal;
+
+    auto& segments = m_segments_choice1.get();
+    int nSegments  = segments.GetSegmentCount();
+
+    bool hasSymmetryPlane = GetNumberOfEdges(segments.GetSegment(1).GetEndWire()) > 1;
+
+    std::array<std::string, 4> names = {loft->Name(), "symmetry", "Front", "Rear"};
+
+    if (!CTiglTopoAlgorithms::IsDegenerated(segments.GetSegment(1).GetStartWire())) {
+        nFacesAero--;
+    }
+    if (!CTiglTopoAlgorithms::IsDegenerated(segments.GetSegment(nSegments).GetEndWire())) {
+        nFacesAero--;
+    }
+
+    int facesPerSegment = nFacesAero / nSegments;
+    int iFaceTotal      = 0;
+    int nSymmetryFaces  = hasSymmetryPlane ? 1 : 0;
+
+    for (int iSegment = 0; iSegment < nSegments; ++iSegment) {
+        for (int iFace = 0; iFace < facesPerSegment - nSymmetryFaces; ++iFace) {
+            loft->FaceTraits(iFaceTotal++).SetName(names[0].c_str());
+        }
+        for (int iFace = 0; iFace < nSymmetryFaces; ++iFace) {
+            loft->FaceTraits(iFaceTotal++).SetName(names[1].c_str());
+        }
+    }
+
+    // Front and rear caps
+    int iFace = 2;
+    for (; iFaceTotal < nFacesTotal; ++iFaceTotal) {
+        loft->FaceTraits(iFaceTotal).SetName(names[iFace++].c_str());
+    }
```

Comment:
Have you checked, wether this implementation works for the fromSimpleParameters method as well?

Reply:
I only used SetFaceTraits for the segment-based vessels. The vessels then just had the uIDs as names.

Instead of a general SetFaceTraits, I can simply add two methods SetFaceTraitsFromSegments and SetFaceTraitsFromParams, where the latter produces `Dome` and `Cylinder`. I want to avoid a deeper logic, because if the number of faceTraits is for example 6, this could either be due to the the radii at torispherical dome, or the polar opening cap for isotensoid. Maybe we name the cylindrical lpart `Cylinder`, and the remaining surfaces `Dome` (I guess a tank must never be open). In the area computation I just exclude `symmetry` then. 

![grafik](https://github.com/user-attachments/assets/ef22fe8a-37ab-46c4-b46b-b6334226daa5)